File: app/src/main/java/com/ivanovsky/passnotes/presentation/StartActivity.java
package com.ivanovsky.passnotes.presentation;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.ivanovsky.passnotes.presentation.unlock.UnlockActivity;

public class StartActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        startActivity(UnlockActivity.Companion.createStartIntent(this));
        finish();
    }
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/presentation/StartActivity.kt
package com.ivanovsky.passnotes.presentation

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.ivanovsky.passnotes.presentation.main.MainActivity
import com.ivanovsky.passnotes.presentation.main.MainScreenArgs

class StartActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val nextIntent = MainActivity.createStartIntent(
            this,
            MainScreenArgs(ApplicationLaunchMode.NORMAL)
        )
        startActivity(nextIntent)
        finish()
    }
}